eclipse-sirius/sirius-components
# [1266] Add support for styling on barChart and pieChart widgets

This Pull Request addresses the #1266 enhancement. 
It introduce the styling capability for bar-chart and pie-chart widget in the form representation.

# Pull request template

## General purpose
What is the main goal of this pull request?

## Project management

## Architectural decision records (ADR)

## Dependencies

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- Hooks are properly typed:

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose

### Architecture

## Review

### How to test this PR?
[View.zip](https://github.com/eclipse-sirius/sirius-components/files/8927691/View.zip)

- Import the attached View model
- Create a new Robot sample (based on flow dsl)
- Create a new Form Representation based on the Form Description provided by the attached View model
- It should look like this:
<img src="https://user-images.githubusercontent.com/7601213/174303598-68e06a72-9051-465f-848e-9b9e83eda462.png" alt="charts" width="250"/>

- Activate the conditional styles on the bar-chart and pie-chart (change aql:false to aql:true)
- Refresh the form representation:
<img src="https://user-images.githubusercontent.com/7601213/174304233-3715c83e-034a-49cd-aa29-9fa9cf0ee99e.png" alt="charts" width="250"/>
